api_v1/endpoints.py

```diff
@@ -0,0 +1,258 @@
+
+            data = User.decode_token(access_token)
+            if not isinstance(data, str):
+                user_id = data
+            else:
+                response = {
+                    'message': data
+                }
+                return response, 401
+
+            return funct(*args, user_id, **kwargs)
+    wrapper.__doc__ = funct.__doc__
+    wrapper.__name__ = funct.__name__
+    return wrapper
+
+sh_ns = Namespace(
+    "Shoppinglist Endpoints",
+    description="Operations related to Shoppinglists and Items",
+    path="/"
+)
+
+shoppinglist_model = sh_ns.model('ShoppingList', {
+    'name': fields.String()
+})
+
+
+@sh_ns.header(
+    "Authorization",
+    description="JWT token for authenticationg users",
+    required=True
+)
+@sh_ns.route("/shoppinglists")
+class Shoppinglists(Resource):
+    """Class to handle multiple shoppinglists"""
+
+    @sh_ns.expect(shoppinglist_model)
+    @token_required
+    def post(self, user_id):
+        """
+            Handle posting of new shoppinglists for an authorized user.
+            Resource Url --> /api/v1/shoppinglists
+        """
+        parser = reqparse.RequestParser()
+
+        parser.add_argument(
+            'name',
+            required=True,
+            help="Required and must be a string"
+        )
+
+        args = parser.parse_args()
+        name = args['name']
+
+        if not re.match("^[-a-zA-Z0-9_\\s]*$", name):
+            response = {
+                "message": "No special characters for shopping list names",
+                "shoppinglist": "null"
+            }
+            return response, 400
+
+        existing_shoppinglist = Shoppinglist.query.filter(
+            func.lower(Shoppinglist.name) == name.lower(),
+            Shoppinglist.created_by == user_id
+        ).first()
+
+        if existing_shoppinglist:
+            response = {
+                "message": "Shopping list already exists!",
+                "shoppinglist": "null"
+            }
+            return response, 400
+
+        try:
+            shoppinglist = Shoppinglist(name=name, created_by=user_id)
+            shoppinglist.save()
+            data = shoppinglist.serialize()
+            sh_json = json.dumps(
+                data, default=datetimeconverter, sort_keys=True
+            )
+            response = {
+                "message": "Shopping List created",
+                "shoppinglist": json.loads(sh_json)
+            }
+            return response, 201
+        except Exception as e:
+            response = {
+                'message': str(e)
+            }
+            return response, 500
+
+    @token_required
+    def get(self, user_id):
+        """
+            Handle getting of all shoppinglists for an authorized user.
+            Resource Url --> /api/v1/shoppinglists
+        """
+        shoppinglists = Shoppinglist.query.filter_by(created_by=user_id)
+        results = []
+
+        for shoppinglist in shoppinglists:
```

Comment:
I have seen that you have used backref and lazy=dyamic in your models. You may want to consider now bringing your shopping list items when you perform a GET shopping lists or GET shoppinglist. Something to think about